docs/api/v1.md

```diff
@@ -0,0 +1,383 @@
+ - `400`: Invalid function
+ - `404`: Function not found
+
+
+## Executions
+An execution represents a chunk of program state and output produced as the program is run. There are two types of Executions--`function`, which is a fully traceable, debuggable execution, and `process`, which only contains `stdout` trace objects.
+
+
+�Function� executions are produced when the process hits a breakpoint or the user executes a standalone function.
+�Process� executions are produced by the `/debug/execute` endpoint. If the execution enters a function with a breakpoint, it terminates and creates a new �function� execution, setting the `nextExecution` field to point to the new function execution.
+
+
+### `GET/debug/executions`
+Get a list of executions since we attached to this process. The server may delete executions data from the current process at any time.
+
+
+Responses:
+ - `200`
+    - Returns {[Execution](#execution)[]}
+
+
+### `GET /debug/executions/:executionId`
+Get information about an execution status.
+
+
+Path parameters:
+ - `executionId`: Execution id
+
+
+Responses:
+ - `200`
+    - Returns {[Execution](#execution)}
+ - `400`: Invalid execution id
+    - Returns {[Error](#error)]}
+ - `404`: Execution id not found
+    - Returns {[Error](#error)]}
+
+
+### `GET /debug/executions/:executionId/trace`
+Get trace data for the execution.
+
+
+Path parameters:
+ - `executionId`: Execution id
+
+
+Responses:
+ - `200`
+    - Returns a stream of {[Trace](#trace)[]}. The body begins with an open bracket `[` on its own line. Each Trace is on its own line. If the execution is still `executing`, new lines will be streamed as they are produced. When the execution is finished the last Trace object has `type=end` and describes how the execution terminated. The final line of the body contains `]`, then the response is closed.
+ - `400`: Invalid execution id
+    - Returns {[Error](#error)]}
+ - `404`: Execution id not found
+    - Returns {[Error](#error)]}
+
+
+### `POST /debug/executions/:executionId/stop`
+Halts a long running Execution.
+
+
+Responses:
+ - `200`: Execution stopped, or was already stopped or finished
+    - Returns {[Execution](#execution)}
+ - `400`: Invalid execution id
+    - Returns {[Error](#error)]}
+ - `404`: Execution id not found
+    - Returns {[Error](#error)]}
+
+
+## Object Definitions
+
+
+### Error
+```
+{
+   code {integer}: Unique code identifying this error type
+   name {string}: Unique human readable name for this error type
+   message {string}: Human readable error message
+   data {object}: Data specific to this error. See [Error Codes](#error-codes) for specifics.
+}
+```
+
+
+
+
+### File
+```
+{
+    name {string}: Name of file
+    path {string}: Path to file
+    type {string}: File type ("file" or "directory")
+    contents: {object[]}:
+        If type="file", undefined
+        If type="directory", an array of Files inside the directory. Directories have "contents" field set to undefined.
+}
```

Comment:
I will clarify the wording--it should say "subdirectories"--i.e. we don't recursively return the entire directory tree--if there are subdirectories, they don't have "contents" set at all, and the client has to make another call to get the contents of each subdirectory. We might want to add a query parameter with the option of getting the entire sub-tree though.